src/debug/dart_debug_impl.ts

```diff
@@ -805,6 +810,32 @@ export class DartDebugSession extends DebugSession {
 		return null;
 	}
 
+	private async pollForMemoryUsage(): Promise<void> {
+		if (!this.childProcess || this.childProcess.killed)
+			return;
+
+		const result = await this.observatory.getVM();
+		const vm = result.result as VM;
+
+		const promises: Array<Promise<DebuggerResult>> = [];
+
+		const isolatePromises = vm.isolates.map((isolateRef) => this.observatory.getIsolate(isolateRef.id));
+		const isolatesResponses = await Promise.all(isolatePromises);
+		const isolates = isolatesResponses.map((response) => response.result as VMIsolate);
+
+		let current = 0;
+		let total = 0;
+
+		for (const isolate of isolates) {
+			current += isolate._heaps.new.used + isolate._heaps.new.external;
```

Comment:
We also need to add in the data from the `old` heap - that's the new generation (young) objects, and the longer lived objects, that have been promoted to the old heap.

Reply:
Done in bc4a42d though it didn't seem to change the numbers much.